cni/log/logger.go

```diff
@@ -0,0 +1,44 @@
+package log
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+type Config struct {
+	Level       zapcore.Level
+	LogPath     string
+	MaxSizeInMB int
+	MaxBackups  int
+	Name        string
+}
+
+var Logger *zap.Logger
+
+func New(cfg *Config) (func(), error) {
+	Logger = newFileLogger(cfg)
```

Comment:
i know today cni logs only to file... if in future, if we wanted to log to file and also send to logcollector/geneva, how can we expand this?

Reply:
Zap allows to register hooks, so you don't need to further instrument the code for that extra logging. Whatever you log to zap, will be logged to geneva through a hook, that can be used to do the proper transformations and implementation
That's why we needed to go to Zap in the first place